cmd/wallet-sdk-js/walletsdk/agent.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright Gen Digital Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package walletsdk implements simplified interface to interop with js.
+package walletsdk
+
+import (
+	"fmt"
+
+	"github.com/hyperledger/aries-framework-go/component/models/did"
+	arieskms "github.com/hyperledger/aries-framework-go/spi/kms"
+
+	"github.com/trustbloc/wallet-sdk/pkg/api"
+	"github.com/trustbloc/wallet-sdk/pkg/did/creator"
+	"github.com/trustbloc/wallet-sdk/pkg/did/resolver"
+	"github.com/trustbloc/wallet-sdk/pkg/localkms"
+)
+
+// Agent is facade around wallet sdk functionality. Provides simplified interface to interop with js.
+type Agent struct {
+	keyWriter   api.KeyWriter
+	crypto      api.Crypto
+	didResolver api.DIDResolver
+}
+
+// NewAgent creates new Agent.
+func NewAgent(didResolverURI string, keyStore arieskms.Store) (*Agent, error) {
+	localKMS, err := localkms.NewLocalKMS(localkms.Config{Storage: keyStore})
+	if err != nil {
+		return nil, fmt.Errorf("fail to create local kms: %w", err)
+	}
+
+	agent := &Agent{
+		keyWriter: localKMS,
+		crypto:    localKMS.GetCrypto(),
+	}
+
+	if didResolverURI != "" {
```

Comment:
Even if didResolverURI is empty, we still need to create a DID resolver (just don't set the WithResolverServerURI option in that case).